### PR TITLE
Events: Migrate old timeRangeUpdate event to new event system to get rid of console.log warning

### PR DESCRIPTION
**Pull request by torkelo**
Tired of seeing "DashboardModel.off is deprecated" in the console

### CODE REVIEW

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx [hunk 1]
```diff
@@ -14,23 +13,23 @@ import { TimePickerWithHistory } from 'app/core/components/TimePicker/TimePicker
...
 
 export class DashNavTimeControls extends Component<Props> {
+  sub?: Unsubscribable;
+
   componentDidMount() {
-    // Only reason for this is that sometimes time updates can happen via redux location changes
-    // and this happens before timeSrv has had chance to update state (as it listens to angular route-updated)
-    // This can be removed after timeSrv listens redux location
-    this.props.dashboard.on(CoreEvents.timeRangeUpdated, this.triggerForceUpdate);
+    this.sub = this.props.dashboard.events.subscribe(TimeRangeUpdatedEvent, () => this.forceUpdate());
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx:28]: Is the previous comment not valid anymore? I think it was great.

File: public/app/features/dashboard/components/SubMenu/DashboardLinks.tsx [hunk 2]
```diff
@@ -21,11 +21,8 @@ export const DashboardLinks: FC<Props> = ({ dashboard, links }) => {
   const [, forceUpdate] = useReducer((x) => x + 1, 0);
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/SubMenu/DashboardLinks.tsx:21]: nit: Not part of this PR, but I think I've seen an `useForceUpdate` hook somewhere

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 3]
```diff
@@ -148,7 +148,7 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
       const prevUrlParams = prevProps.queryParams;
       const urlParams = this.props.queryParams;
 
-      if (urlParams?.from !== prevUrlParams?.from && urlParams?.to !== prevUrlParams?.to) {
+      if (urlParams?.from !== prevUrlParams?.from || urlParams?.to !== prevUrlParams?.to) {
```
Comment (hugohaggmark) [inline public/app/features/dashboard/containers/DashboardPage.tsx:151]: Whoa, great catch. Maybe we should have a unit test for this?

**Review by hugohaggmark — APPROVED** [review]
LGMT 🎉

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx:28]: it's not really valid anymore as it pertained to redux location and angular route events.

Comment (torkelo) [reply public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx:28]: and we have none of those after v8

File: public/app/features/dashboard/components/SubMenu/DashboardLinks.tsx (hunk 2 above)
Comment (torkelo) [reply public/app/features/dashboard/components/SubMenu/DashboardLinks.tsx]: ah, good catch updated to useForceUpdate

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx [hunk 4]
```diff
@@ -14,23 +13,23 @@ import { TimePickerWithHistory } from 'app/core/components/TimePicker/TimePicker
...
+import { ShiftTimeEvent, ShiftTimeEventPayload, TimeRangeUpdatedEvent, ZoomOutEvent } from '../../../../types/events';
+import { Unsubscribable } from 'rxjs';
 
 export interface Props {
   dashboard: DashboardModel;
   onChangeTimeZone: (timeZone: TimeZone) => void;
 }
 
 export class DashNavTimeControls extends Component<Props> {
+  sub?: Unsubscribable;
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx]: updated to use private
